gammapy/spectrum/models.py

```diff
@@ -574,30 +577,131 @@ class TableModel(SpectralModel):
         try:
             self.unit = self.values.unit
-            logy = np.log10(self.values.value)
+            if scale_logy is True:
+                y = np.log10(self.values.value)
+            else:
+                y = self.values.value
         except AttributeError:
             self.unit = u.Unit('')
-            logy = np.log10(self.values)
+            if scale_logy is True:
+                y = np.log10(self.values)
+            else:
+                y = self.values
+        self.interpy = interp1d(loge,
+                                y,
+                                fill_value=-np.Inf,
+                                bounds_error=False,
+                                kind='cubic')
 
-        self.interplogy = interp1d(loge,
-                                   logy,
-                                   fill_value=-np.Inf,
-                                   bounds_error=False,
-                                   kind='cubic')
+    @classmethod
+    def from_file(cls, filename, param):
+        """A Table containing aborbed values from a XSPEC model
+        as a function of energy.
+        Todo:
+        Format of the file should be described and discussed in
+        https://gamma-astro-data-formats.readthedocs.io/en/latest/index.html
+
+        Parameters
+        ----------
+        filename : `str`
+            File containing the XSPEC model
+        param : float
+            Model parameter value
+        Examples
+        --------
+        Fill table from an EBL model (Franceschini, 2008)
+            from gammapy.spectrum.models import TableModel
+            filename = '$GAMMAPY_EXTRA/datasets/ebl/ebl_franceschini.fits.gz'
+            table_model = TableModel.from_file(filename=filename, param=0.3)
+        """
+        filename = str(make_path(filename))
+
+        # Check if parameter value is in range
+        table_param = Table.read(filename, hdu='PARAMETERS')
+        param_min = table_param['MINIMUM']
+        param_max = table_param['MAXIMUM']
+        assert(param >= param_min and param <= param_max), "Parameter\
+ out of range, param={0}, param_min={1}, param_max={2}".format(param,
+                                                               param_min,
+                                                               param_max)
+        # Get energy values
+        table_energy = Table.read(filename, hdu='ENERGIES')
+        energy_lo = table_energy['ENERG_LO']
+        energy_hi = table_energy['ENERG_HI']
+
+        # Hack while format is not fixed, energy values are in keV
+        energy_bounds = EnergyBounds.from_lower_and_upper_bounds(lower=energy_lo,
+                                                                 upper=energy_hi,
+                                                                 unit=u.keV)
+        energy = energy_bounds.log_centers
+
+        # Get spectrum values (no interpolation, take closest value for param)
+        table_spectra = Table.read(filename, hdu='SPECTRA')
```

Comment:
Is it ok to hardcode the ``hdu`` names? Is this always the same for the format you're using, or is it better to pass the hdu names as kwargs?